src/hotspot/share/jfr/periodic/sampling/jfrThreadSampling.cpp

```diff
@@ -307,6 +352,30 @@ static void drain_enqueued_requests(const JfrTicks& now, JfrThreadLocal* tl, Jav
   assert(!tl->has_enqueued_requests(), "invariant");
 }
 
+
+static void drain_all_enqueued_requests(const JfrTicks& now, JfrThreadLocal* tl, JavaThread* jt, Thread* current) {
+  assert(tl != nullptr, "invariant");
+  assert(jt != nullptr, "invariant");
+  assert(current != nullptr, "invariant");
+  drain_enqueued_requests(now, tl, jt, current);
+#ifdef LINUX
+  if (tl->has_cpu_time_jfr_requests()) {
```

Comment:
You are having all threads traverse over this lock, even though the cpu time sampler is disabled by default. Can it be improved?

Reply:
I don't understand what allocation has to do with anything. I'm talking about code branch layout to avoid having to test "has_cpu_time_jfr_requests()" when we know it will be false by default.

Reply:
Right.

Reply:
I pass it through now.